src/Config/ConfigLocator.php

```diff
@@ -256,29 +259,92 @@ public function addDrushConfig($drushProjectDir)
      * Add any configuration files found around the Drupal root of the
      * selected site.
      *
-     * @param Path to the selected Drupal site
+     * @param $drupalRoot
+     *   Path to the selected Drupal site.
      * @return $this
      */
-    public function addSitewideConfig($siteRoot)
+    public function addDrupalConfig($drupalRoot)
     {
         // There might not be a site.
-        if (!is_dir($siteRoot)) {
+        if (!is_dir($drupalRoot)) {
             return;
         }
 
         // We might have already processed this root.
-        $siteRoot = realpath($siteRoot);
-        if (in_array($siteRoot, $this->siteRoots)) {
+        $drupalRoot = realpath($drupalRoot);
+        if (in_array($drupalRoot, $this->drupalRoots)) {
             return;
         }
 
         // Remember that we've seen this location.
-        $this->siteRoots[] = $siteRoot;
+        $this->drupalRoots[] = $drupalRoot;
+
+        $this->addConfigPaths(self::DRUPAL_CONTEXT, [ dirname($drupalRoot) . '/drush', "$drupalRoot/drush", "$drupalRoot/sites/all/drush" ]);
+        return $this;
+    }
+
+    /**
+     * Add any configuration files found around the multisite directory.
+     *
+     * @param \Drush\SiteAlias\AliasRecord $alias
+     *   Site URI of the multisite.
+     *
+     * @return $this
+     */
+    public function addSiteConfig($alias)
+    {
+        $uri = $alias->uri() ?: 'default';
+
+        // Convert a fqdn to a hostname and look for matching entry in
+        // sites/sites.php.
+        if (filter_var($uri, FILTER_VALIDATE_URL)) {
+            $hostname = StringUtils::convertUriToHostname($uri);
+
+            // If $hostname matches a sites.php mappings, use dir from mapping.
+            if ($dir_name = $this->lookupSiteDirFromHostname($hostname, $alias->root())) {
+                $uri = $dir_name;
+            }
+        }
 
-        $this->addConfigPaths(self::DRUPAL_CONTEXT, [ dirname($siteRoot) . '/drush', "$siteRoot/drush", "$siteRoot/sites/all/drush" ]);
+        // There might not be a site    directory.
+        $site_dir = $alias->root() ."/sites/$uri";
+        if (!is_dir($site_dir)) {
+            return;
+        }
+
+        // We might have already processed this site.
+        if (in_array($site_dir, $this->siteDirs)) {
+            return;
+        }
+
+        // Remember that we've seen this site.
+        $this->siteDirs[] = $site_dir;
+
+        $this->addConfigPaths(self::SITE_CONTEXT, [ "$site_dir", "$site_dir/drush" ]);
         return $this;
     }
 
+    /**
+     * Lookup a site's directory via the sites.php file given a hostname.
+     *
+     * @param $hostname
+     *   The hostname of a site. May be converted from URI.
+     *
+     * @return $drupalRoot
+     *   The directory associated with that hostname.
+     */
+    public function lookupSiteDirFromHostname($hostname, $drupalRoot) {
```

Comment:
Not sure where this belongs. I can move to StringUtils, though it's not strictly doing string manipulation.